server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -187,6 +197,17 @@ public static boolean isRemoteStoreClusterStateEnabled(Settings settings) {
             && isRemoteClusterStateAttributePresent(settings);
     }
 
+    private static boolean isRemoteRoutingTableAttributePresent(Settings settings) {
+        return settings.getByPrefix(Node.NODE_ATTRIBUTES.getKey() + REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY)
+            .isEmpty() == false;
+    }
+
+    public static boolean isRemoteRoutingTableEnabled(Settings settings) {
+        return FeatureFlags.isEnabled(REMOTE_ROUTING_TABLE_EXPERIMENTAL)
+            && RemoteRoutingTableService.REMOTE_ROUTING_TABLE_ENABLED_SETTING.get(settings)
+            && isRemoteRoutingTableAttributePresent(settings);
+    }
```

Comment:
Why do we require two flags?

Reply:
`REMOTE_ROUTING_TABLE_EXPERIMENTAL` is to keep this as an experimental feature which will have to be enabled from JVM options.`REMOTE_ROUTING_TABLE_ENABLED_SETTING` is cluster setting for ease of enabling and disabling.

Reply:
Changed experimental feature to common one for Remote state publication. Keeping the check here in addition to cluster settings check since we want to enable this logic only if publication is enabled as well.